### ISSUE TITLE
interactive_hist doesn't display actual histogram

### ISSUE THREAD

**Issue by Tianmaru** [mpl-extensions/mpl-interactions#293]
### Bug report

**Bug summary**

`interactive_hist` doesn't show the actual histogram.

**Code for reproduction**

```python
import numpy as np
from mpl_interactions import interactive_hist

bot = np.linspace(0, 100, 100)
top = np.linspace(0, 100, 100)


def callback(bot, top):
    return np.random.randn(1000)

interactive_hist(callback, bot=bot, top=top)
```

**Actual outcome** 

Matplotlib window with control sliders, but without actual histogram.

![grafik](https://github.com/user-attachments/assets/982db37a-e416-4ec5-a99c-c29875cef1be)

**Expected outcome**

Matplotlib histogram plot with bot and top control sliders.

**Version Info**

  * Operating system: Debian 12
  * Matplotlib version: 3.9.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): tkagg, qtagg
  * Python version: 3.12.4

**Comment by ianhi** [comment]
Hi @Tianmaru,

Can I asked what context you are runnign this in? e.g.: `ipython`, python script, `python repl`, `jupyter` etc...

I suspect that what is happening is actually the defualt maptlotlib behavior and that mpl-interactiosn is being a bit eager about displaying the control figure.

If you add a plt.show() at the end of the script you will see the histogram

**Comment by Tianmaru** [comment]
Hi @ianhi,
thanks for your answer! I don't know why I didn't try to call plt.show myself. I think I expected the controls and the plot to show up in the same window, below the plot like via IPython? It looks like this now:

![grafik](https://github.com/user-attachments/assets/ae308819-b69b-4249-8ad6-7e3650d7c63e)

**Comment by Tianmaru** [comment]
Oh, and regarding your question: the code is run as a Python Script.

**Comment by Tianmaru** [comment]
Ah, and I noticed something else: when trying to use a `RangeSlide` like described in the [mpl-interactions documentation](https://mpl-interactions.readthedocs.io/en/stable/examples/range-sliders.html), I get

`TypeError: 'tuple object does not support item assignment`.

This seems to be an issue with the `valinit` argument being provided to `RangeSlider` as a tuple in `create_mpl_range_selection_slider`, which is then accessed in `widget's` `set_val`.

## Code Example

```python
import numpy as np
import matplotlib.pyplot as plt
from mpl_interactions import interactive_hist

xrange = ("range", 0, 100, 1)


def callback(xrange):
    return np.random.randn(1000)


interactive_hist(callback, xrange=xrange)
plt.show()
```

**Comment by ianhi** [comment]
ahh good catch thank you! That is indeed a bug. I hope to have time today to fix it and will publish a new release as soon as i do.

**Comment by ianhi** [comment]
@Tianmaru just fixed this in #294 and it is released in version `0.24.2` which is available on pypi now and should be on conda-forge before too long.

One other thing to note, the documentation was slightly incorrect in how it described the shorthand for generating a slider.

Where it said: `("range", min, max, [step]) ` it should have read `("range", min, max, [num])` i.e. using `linspace` not `arange`. 

So to get the intended behavior in your last post you would need to write: `xrange = ("range", 0, 100, 100)` or alternatively `xrange = ("range", np.arange(0,100,1))`

**Comment by ianhi** [comment]
Thank you for the helpful bug report!
